lib/conrad/emitter_queue.rb

```diff
@@ -0,0 +1,52 @@
+module Conrad
+  class EmitterQueue
+    include Singleton
+
+    attr_reader :background
+    attr_accessor :logger
+
+    def initialize
+      Thread.abort_on_exception = true
```

Comment:
We should *absolutely* not do this as this sets this up globally, potentially and unexpectedly changing anybody's behavior. We'll need to do something like this each time we start up a Thread:

```ruby
Thread.new do
  Thread.current.abort_on_exception = true
  # whatever else the Thread does
end
```

Reply:
oh nice. i wasn't sure the best way to do this.

Reply:
I was really hoping that `Thread.new` would accept kwargs to set it up. Sadly, no. And doing it after initialization means we might be changing a dead thread :/

Reply:
Is there harm in changing a dead thread? it doesn't really matter, right?

Reply:
It matters if said Thread hits an error before we've told it to re-raise them. The Thread starts running as soon as the `do...end` block finishes evaluating, so it's possible to get an unpleasant race condition there.

Reply:
I think we're talking about an outdated diff. Is what you're describing still an issue here:
```ruby
    def start_thread
      @thread ||= Thread.new do
        Thread.current.abort_on_exception = true
        loop do
          emit!
          break unless Conrad::EmitterQueue.instance.background
        end
      end
    rescue e
      logger.error(e)
      @thread = nil
      start_thread
    end
```

Reply:
It is outdated, but you left a comment with a question so I replied in the github thread 😬